pandas/core/strings.py

```diff
@@ -109,9 +118,48 @@ def cat_safe(list_of_columns: List, sep: str):
 
 def _na_map(f, arr, na_result=np.nan, dtype=object):
     # should really _check_ for NA
+    if is_extension_array_dtype(arr.dtype):
+        arr = extract_array(arr, extract_numpy=True)
```

Comment:
```suggestion
        arr = extract_array(arr)
```

? 
`_map_ea` should always get a extension array, never a numpy array